pkg/operator/edgefs/cluster/prepare/prepare.go

```diff
@@ -0,0 +1,261 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package prepare for the Edgefs manager.
+package prepare
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/coreos/pkg/capnslog"
+	rookalpha "github.com/rook/rook/pkg/apis/rook.io/v1alpha2"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	batch "k8s.io/api/batch/v1"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var logger = capnslog.NewPackageLogger("github.com/rook/rook", "edgefs-op-prepare")
+
+const (
+	appName                   = "host-prepare"
+	defaultServiceAccountName = "rook-edgefs-cluster"
+	etcVolumeName             = "etc-volume"
+)
+
+// Cluster is the edgefs prepare manager
+type Cluster struct {
+	Namespace      string
+	Version        string
+	serviceAccount string
+	placement      rookalpha.Placement
+	context        *clusterd.Context
+	resources      v1.ResourceRequirements
+	ownerRef       metav1.OwnerReference
+	ctx            context.Context
+}
+
+// New creates an instance of the prepare
+func New(
+	context *clusterd.Context, namespace, version string,
+	serviceAccount string,
+	placement rookalpha.Placement,
+	resources v1.ResourceRequirements,
+	ownerRef metav1.OwnerReference,
+) *Cluster {
+
+	if serviceAccount == "" {
+		// if the service account was not set, make a best effort with the example service account name since the default is unlikely to be sufficient.
+		serviceAccount = defaultServiceAccountName
+		logger.Infof("setting the prepare pods to use the service account name: %s", serviceAccount)
+	}
+
+	return &Cluster{
+		context:        context,
+		Namespace:      namespace,
+		serviceAccount: serviceAccount,
+		placement:      placement,
+		Version:        version,
+		resources:      resources,
+		ownerRef:       ownerRef,
+	}
+}
+
+// Start the prepare instance
+func (c *Cluster) Start(rookImage string, nodeName string) error {
+	logger.Infof("start running prepare pods")
+
+	// start the deployment
+	job := c.makeJob(appName, c.Namespace, rookImage, nodeName)
+	if _, err := c.context.Clientset.BatchV1().Jobs(c.Namespace).Create(job); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return fmt.Errorf("failed to create %s job. %+v", appName, err)
+		}
+		logger.Infof("%s daemonset already exists", appName)
+	} else {
+		logger.Infof("%s job started", appName)
+	}
+
+	err := c.waitJob(job)
+	if err != nil {
+		logger.Warningf("Job %s waiting failed. %+v", job.ObjectMeta.Name, err)
+	}
+
+	err = c.deleteJob(job)
+	if err != nil {
+		logger.Warningf("Failed to delete job %s due. %+v", job.ObjectMeta.Name, err)
+	}
+
+	return nil
+}
+
+func (c *Cluster) makeJob(name, clusterName, rookImage string, nodeName string) *batch.Job {
+
+	volumes := []v1.Volume{
+		{
+			Name: etcVolumeName,
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: "/etc",
```

Comment:
Why is there a need to modify the hosts `/etc` directory (read only is not set)?

Reply:
Yeah, but why? Do you need to edit the hosts file? sysctl.conf? Another config under `/etc`?

Would be good to document what is changed as this means a removal of the EdgeFS cluster possibly means manual user interaction to change those edits back.

Reply:
Agree. It needs to be better documented, especially how/what needs to be reverted on the node. In this code modification, we do provide skipHostPrepare option which user can set in terms of to avoid touching host config altogether. EdgeFS relying on Linux UDP/IP stack, which needs to be tuned in terms of to use all the variety of use cases it's being designed for. We do however already provide documentation in quickstart doc, not enough - we will need to add reference to it from Cluster CRD also!